src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -46,38 +46,88 @@ public bool CheckRequestAllowsCaching()
         }
 
         // Only QueryString is treated as case sensitive
-        // GET;HTTP://MYDOMAIN.COM:80/PATHBASE/PATH?QueryString
+        // GET;HTTP://MYDOMAIN.COM:80/PATHBASE/PATH?QueryString;VaryBy
```

Comment:
Seems like most implementations do not consider scheme, host and port in the key generation. I think that might be a good default to start with.